src/pages/admin/donate/components/generic-details/GenericDetails.tsx

```diff
@@ -244,6 +244,7 @@ export function GenericDetails<T extends { id: number } & FieldValues>({
                                 className="generic-details btn-add"
                                 onClick={handleAdd}
                                 buttonStyle={primaryAddButton ? 'primary' : 'secondary'}
+                                disabled={isAddButtonDisabled || isParentAddFormVisible || isAddFormVisible}
```

Comment:
explanation in comment above

```suggestion
                                disabled={isDisabled || isAddButtonDisabled || isParentAddFormVisible || isAddFormVisible}
```